src/components/dialogs/generator-modification-dialog.js

```diff
@@ -113,12 +114,13 @@ const GeneratorModificationDialog = ({
             : { id: '' };
     }, [formValues]);
 
+    console.info(equipmentOptions);
     const [generatorInfos, generatorIdField] = useAutocompleteField({
         label: 'ID',
         validation: { isFieldRequired: true },
         inputForm: inputForm,
         formProps: filledTextField,
-        values: equipmentOptions?.sort(compareById),
+        values: equipmentOptions ? equipmentOptions?.sort(compareById) : [],
```

Comment:
Why repeat equipmentOptions, when using `?.` ? `equipmentOptions?.sort(compareById) ?? []` should do